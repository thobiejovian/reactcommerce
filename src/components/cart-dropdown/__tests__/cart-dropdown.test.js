import React from 'react';
import { shallow, mount } from 'enzyme';
import {Provider} from "react-redux";
import {BrowserRouter as Router} from "react-router-dom";
import {store} from "../../../redux/store";

import { CartDropdown }  from '../cart-dropdown.component';
import CartItem from '../../cart-item/cart-item.component';
import { toggleCartHidden } from '../../../redux/cart/cart.actions';

import cartItemsMocks from "../../../../mocks/cartItemsMocks.json";
describe('CartDropdown component', () => {
    let wrapper;
    let mockHistory;
    let mockDispatch;
    const mockCartItems = cartItemsMocks;

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        };

        mockDispatch = jest.fn();

        const mockProps = {
            cartItems: mockCartItems,
            history: mockHistory,
            dispatch: mockDispatch
        };

        wrapper = mount(
            <Provider store={store}>
                <Router>
                    <CartDropdown {...mockProps} />
                </Router>
            </Provider>);
    });

    it('should render CartDropdown component without error', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should call history.push when button is clicked', () => {
        wrapper.find('CustomButton').simulate('click');
        expect(mockHistory.push).toHaveBeenCalled();
    });

    it('should render an equal number of CartItem components as the cartItems prop', () => {
        expect(wrapper.find('CartItem').length).toEqual(mockCartItems.length);
    });

    it('should render empty message if cartItems is empty', () => {
        const mockProps = {
            cartItems: [],
            history: mockHistory,
            dispatch: mockDispatch
        };
        const newWrapper = mount(
            <Provider store={store}>
                <Router>
                    <CartDropdown {...mockProps} />
                </Router>
            </Provider>);
        expect(newWrapper.exists('.empty-message')).toBe(true);
        expect(newWrapper.debug()).toMatchSnapshot();
    });
});