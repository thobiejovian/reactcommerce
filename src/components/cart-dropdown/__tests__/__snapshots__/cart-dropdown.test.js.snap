// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown component without error 1`] = `
"<Provider store={{...}}>
  <BrowserRouter>
    <Router history={{...}}>
      <CartDropdown cartItems={{...}} history={{...}} dispatch={[Function: mockConstructor]}>
        <div className=\\"cart-dropdown\\">
          <div className=\\"cart-items\\">
            <CartItem item={{...}}>
              <div className=\\"cart-item\\">
                <img src=\\"https://i.ibb.co/ZYW3VTp/brown-brim.png\\" alt=\\"item\\" />
                <div className=\\"item-details\\">
                  <span className=\\"name\\">
                    Brown Brim
                  </span>
                  <span className=\\"price\\">
                     x €
                    25
                  </span>
                </div>
              </div>
            </CartItem>
            <CartItem item={{...}}>
              <div className=\\"cart-item\\">
                <img src={[undefined]} alt=\\"item\\" />
                <div className=\\"item-details\\">
                  <span className=\\"name\\" />
                  <span className=\\"price\\">
                     x €
                  </span>
                </div>
              </div>
            </CartItem>
            <CartItem item={{...}}>
              <div className=\\"cart-item\\">
                <img src={[undefined]} alt=\\"item\\" />
                <div className=\\"item-details\\">
                  <span className=\\"name\\" />
                  <span className=\\"price\\">
                     x €
                  </span>
                </div>
              </div>
            </CartItem>
          </div>
          <CustomButton onClick={[Function: onClick]}>
            <button className=\\"  custom-button\\" onClick={[Function: onClick]}>
              GO TO CHECKOUT
            </button>
          </CustomButton>
        </div>
      </CartDropdown>
    </Router>
  </BrowserRouter>
</Provider>"
`;

exports[`CartDropdown component should render empty message if cartItems is empty 1`] = `
"<Provider store={{...}}>
  <BrowserRouter>
    <Router history={{...}}>
      <CartDropdown cartItems={{...}} history={{...}} dispatch={[Function: mockConstructor]}>
        <div className=\\"cart-dropdown\\">
          <div className=\\"cart-items\\">
            <span className=\\"empty-message\\">
              Your cart is empty
            </span>
          </div>
          <CustomButton onClick={[Function: onClick]}>
            <button className=\\"  custom-button\\" onClick={[Function: onClick]}>
              GO TO CHECKOUT
            </button>
          </CustomButton>
        </div>
      </CartDropdown>
    </Router>
  </BrowserRouter>
</Provider>"
`;
